src/core/components.py

```diff
@@ -793,3 +817,138 @@ def get_base_component_dict(self):
     def get_components(self):
         yield from super().get_components()
         yield from self.components
+
+
+class Panel(Component):
+    """A panel which can contain components."""
+
+    __slots__ = [
+        "components",
+        "width",
+        "is_card",
+        "style",
+    ]
+
+    def __init__(
+        self,
+        components,
+        width=12,
```

Comment:
I think I understand the idea of separating the concerns of height and with meaning that you can enforce regularity along a row. However, as soon as you want anything more complex than just regular hight columns in a row things become complex very quickly consider something like this


```
-----------------------------------------------------
|                   |                               |
|                   |                               |
|-------------------|                               |
|                   |                               |
|                   |                               |
|----------------------------------------------------
|                                                   |
|                                                   |
|                                                   |
|                                                   |
|                                                   |
-----------------------------------------------------
````
You'd need a 
```
[
    Row(
        components=[
            Column(
                components=[
                    Row(), 
                    Row()
                ]
            )
            Column()
        ]
    ),
    Row()
]
```

Reply:
Yep it is something we will likely need in future when we need to do more complicated rows, but for this story is out of scope.

Reply:
My point is another way of doing it would be something like this would be by having something with height and width on the same object then using some fractional co-ordinate system it becomes something like:
```
[
    Cell(origin=(0,0), height=0.25, width=0.3),
    Cell(origin=(0,0.25), height=0.25, width=0.3),
    Cell(origin=(0.3,0), height=0.5, width=0.7), 
    Cell(origin=(0,0.5), height=0.5, width=1),
]
```

Reply:
Yeah I get where you're coming from, but this story is just to get the basic structure in place so we can support a fairly straightforward setup with components in specific containers.